src/main/java/gui/view/MainPageController.java

```diff
@@ -0,0 +1,159 @@
+     */
+    private void updateListView(JFXListView<HBox> listView, Map<String, String> map) {
+        listView.getItems().forEach(i -> i.setVisible(false));
+        listView.getItems().clear();
+        for (Map.Entry<String, String> taskInfo : map.entrySet()) {
+            Label taskName = new Label(taskInfo.getKey());
+            Label deadline = new Label(taskInfo.getValue());
+            taskName.setFont(new Font(labelFontSize));
+            deadline.setFont(new Font(labelFontSize));
+            taskName.setMinWidth(550);
+            taskName.setMaxWidth(550);
+            HBox task = new HBox(taskName, deadline);
+            listView.getItems().add(task);
+        }
+        listView.refresh();
+
+    }
+
+    /**
+     * Note: removes visibility of all previously rendered labels in the list view before flushing it out.
+     * @param eventListView the list view to update
+     * @param map a mapping of name and time for today's relevant items
+     */
+    private void updateEventListView(JFXListView<HBox> eventListView, Map<String, List<String>> map) {
+        eventListView.getItems().forEach(i -> i.setVisible(false));
+        eventListView.getItems().clear();
+        for (Map.Entry<String, List<String>> eventInfo : map.entrySet()) {
+            Label eventName = new Label(eventInfo.getKey());
+            Label startTime = new Label("Start Time is " + eventInfo.getValue().get(0));
+            Label endTime = new Label("End Time is " + eventInfo.getValue().get(1));
+            eventName.setFont(new Font(labelFontSize));
+            startTime.setFont(new Font(labelFontSize));
+            endTime.setFont(new Font(labelFontSize));
+            eventName.setMinWidth(200);
+            eventName.setMaxWidth(200);
+            startTime.setMinWidth(300);
+            startTime.setMaxWidth(300);
+
+            HBox task = new HBox(eventName, startTime, endTime);
+            eventListView.getItems().add(task);
+        }
+        eventListView.refresh();
+
+    }
+
+
+    /**
+     * Method listening to updates from the {@link MainPageViewModel}.
+     * @param readers the lists of updated readers
+     */
+    private void acceptReaders(MainPageViewModel.Readers readers) {
+        updateListView(taskListView, formatTaskReaders(readers.taskReaders));
+        updateEventListView(eventListView, formatEventReaders(readers.eventReaders));
+    }
+
+    private Map<String, String> formatTaskReaders(List<TaskReader> taskReaders) {
+        Map<String, String> taskMap = new HashMap<>();
+        if (taskReaders.isEmpty()){
+            taskMap.put("No Scheduled Tasks", null);
+        }
+        for (TaskReader task: taskReaders) {
+            String taskName = task.getName();
+            String deadline = formatDuration(task.getDeadline());
+            taskMap.put(taskName, deadline);
+        }
+        return taskMap;
+    }
+
+    private String formatDuration(LocalDateTime localDateTime) {
```

Comment:
In my understanding, view is just receiving something to display? So probably move the converting LocalDateTime to String to view model?